Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1030,4 +1030,49 @@ bool check_that_a_key_could_be_injected_into_a_value(JS::Realm& realm, JS::Value
     return value.is_object() || MUST(value.is_array(realm.vm()));
 }
 
+// https://w3c.github.io/IndexedDB/#fire-an-error-event
+void fire_an_error_event(JS::Realm& realm, GC::Ref<IDBRequest> request)
+{
+    // 1. Let event be the result of creating an event using Event.
+    // 2. Set event’s type attribute to "error".
+    // 3. Set event’s bubbles and cancelable attributes to true.
+    auto event = DOM::Event::create(realm, HTML::EventNames::error, { .bubbles = true, .cancelable = true });
+
+    // 4. Let transaction be request’s transaction.
+    auto transaction = request->transaction();
+
+    // 5. Let legacyOutputDidListenersThrowFlag be initially false.
+    bool legacy_output_did_listeners_throw_flag = false;
+
+    // 6. If transaction’s state is inactive, then set transaction’s state to active.
+    if (transaction->state() == IDBTransaction::TransactionState::Inactive)
+        transaction->set_state(IDBTransaction::TransactionState::Active);
+
+    // 7. Dispatch event at request with legacyOutputDidListenersThrowFlag.
+    DOM::EventDispatcher::dispatch(request, *event, false, legacy_output_did_listeners_throw_flag);
+
+    // 8. If transaction’s state is active, then:
+    if (transaction->state() == IDBTransaction::TransactionState::Active) {
+        // 1. Set transaction’s state to inactive.
+        transaction->set_state(IDBTransaction::TransactionState::Inactive);
+
+        // 2. If legacyOutputDidListenersThrowFlag is true, then run abort a transaction with transaction and a newly created "AbortError" DOMException and terminate these steps.
+        //    This is done even if event’s canceled flag is false.
+        if (legacy_output_did_listeners_throw_flag) {
+            abort_a_transaction(*transaction, WebIDL::AbortError::create(realm, "An error occurred"_string));
```

Comment:
This seems like a pretty lame error message. Is there anything else we can say? "Error event interrupted by exception" ? or something. makes me wonder what kind of errors other implementations throw when this `legacyOutputDidListenersThrowFlag` is involved (in other WebIDL interfaces as well)

Reply:
I cant make heads or tails of the chromium source, but from what i could tell, they dont have this flag at all?